test/e2e/performanceprofile/functests/8_reboot/devices.go

```diff
@@ -0,0 +1,388 @@
+		Expect(err).ToNot(HaveOccurred(), "error creating workload deployment")
+
+		// short timeout: we are on idle cluster
+		wlDp = waitForDeploymentCompleteOrFail("pre reboot", *wlDp, 5*time.Minute)
+		testlog.Infof("deployment %s/%s running", wlDp.Namespace, wlDp.Name)
+
+		// phase3: the reboot, which trigger the very scenario we are after
+
+		// managed, clean restart (e.g. `reboot` command or `systemctl reboot`
+		// - details don't matter as long as this is a managed clean restart).
+		// Power loss scenarios, aka hard reboot, deferred to another test.
+		// intentionally ignoring error. We need to tolerate connection error or disconnect
+		// because the node is rebooting.
+		runCommandOnNodeThroughMCD(node, "reboot", rebootNodeCommandMCD)
+		// this is (likely) a SNO. We need to tolerate connection errors,
+		// because the apiserver is going down as well.
+		// we intentionally use a generous timeout.
+		// On Bare metal reboot can take a while.
+		waitForNodeReadyOrFail("post reboot", targetNode, 20*time.Minute, 3*time.Second)
+
+		// are we really sure? we can't predict if we will have state flapping,
+		// we can't predict if pods go back to containercreating and ideally we
+		// should have no flapping.
+		// Tracking all the state will make the test complex *and fragile*.
+		// The best we can do right now is to let the SNO cool down and check again.
+		testlog.Infof("post reboot: entering cooldown time: %v", rebootCooldownTime)
+		time.Sleep(rebootCooldownTime)
+		testlog.Infof("post reboot: finished cooldown time: %v", rebootCooldownTime)
+
+		// longer timeout. We expect things to be still on flux
+		waitForNodeToReportResourcesOrFail("post reboot", targetNode, sriovDeviceResourceName, 10*time.Minute, 2*time.Second)
+
+		// from now on we assume the node is somehow stable again and we stop swallowing the connection errors
+
+		// what happened to the previous workload? Note we need a long timeout, because we have no means to learn when the SNO is actually settled
+		wlDp = waitForDeploymentCompleteOrFail("post reboot", *wlDp, 30*time.Minute)
+		testlog.Infof("deployment %s/%s running", wlDp.Namespace, wlDp.Name)
+
+		// do we have any pods from the workload deployment on UnexpectedAdmissionError? note it's legit to not have them at all if we got lucky.
+		// but we should have some most of time.
+		pods, err := listPodsByDeployment(testclient.Client, context.TODO(), *wlDp)
+		Expect(err).ToNot(HaveOccurred(), "error checking the pods belonging to deployment %s/%s", wlDp.Namespace, wlDp.Name)
+		if len(pods) > int(*wlDp.Spec.Replicas) {
+			admissionFailed := 0
+			for idx := range pods {
+				if isUnexpectedAdmissionError(pods[idx], sriovDeviceResourceName) {
+					admissionFailed++
+				}
+			}
+			testlog.Infof("workload deployment %s/%s got %d admission errors", wlDp.Namespace, wlDp.Name, admissionFailed)
+		} else {
+			testlog.Infof("!!! workload deployment %s/%s got running at first try !!!", wlDp.Namespace, wlDp.Name)
+		}
```

Comment:
I don't feel too strongly but this seemed a bit complicated to me on the first go. Took me a minute to wrap my head around what's going on here. In general it is okay but perhaps it would be more straightforward to identify the pod name before node reboot happens and query for that specific pod via the API and crosscheck its status? 
Another reason for my suggestion is that we are creating the deployment with one replica so it makes sense to identify the pod that was created as part of the deployment pre-reboot and check for it later. If we had multiple replicas, what you have currently makes more sense.

Reply:
I'm not keen of complex code (simplified a bit this area after your comment). I think it's important to capture this detail to implement a good testcase. The reason why we are doing this here post-reboot are:

1. we should check what happens to the deployment on the recovery path. Depending on unpredictable recovery ordering and cluster load, we may have few attempts from the deployment controller to create the single replica we need, with few failures for UnexpectedAdmissionError.
2. we should check the fate of all the attempts, trying to detect unexpected failures (e.g. we expect all the failures to be caused by the fix, not because other reasons)
3. on top of all the above, actually going running at the first try is legit!

For all these reasons I'm inclined to make the test check all the pods pertaining to the deployment post-reboot. I reckon this implementation of the concept may be suboptimal and I'm open to suggestions and improvements here.

Reply:
Fair enough, let's check for the whole deployment.